### ISSUE TITLE
Support kubectl edit

### ISSUE THREAD

**Issue by deathgaze** [Azure/vscode-kubernetes-tools#904]
Title says it all. I would love to be able to right click on resources and edit their manifests directly in VS Code. VS Code doesn't seem to directly support this at the moment due to it not waiting for exit on standard in properly.

**Comment by itowlson** [comment]
Just to clarify the request, what experience do you envisage here?

1. If you run `kubectl edit` at the command line (outside VS Code), it loads the YAML, and kubectl applies the changes when you close the file?
2. There should be an Edit command on resources in the tree (and via the command palette), which would bring up the YAML a la Load. and would then apply the changes when you:
  a. closed the file?
  b. saved the file?  (i.e. 'save' would be to the k8s resource itself instead of to the filesystem)
  c. issued an explicit Apply Changes command?
3. Something else?

**Comment by deathgaze** [comment]
I would envision it operating exactly as it should on the command line if VS Code were invoked with the `—wait` option on the CLI (I’m not sure if it’s relevant, but git commit messages using VS code as editor work exactly this way). If a manifest has been modified, it should be marked as dirty and the user should be prompted to save it. If a manifest is already saved and clean when it is closed, the manifest should automatically be applied.

**Comment by MurzNN** [comment]
Yes, I also vote for option "b" - on save action ask user to apply modified manifest or not.

**Comment by pWoLiAn** [comment]
cant we do this already by loading the YAML, making changes and applying it using "kubectl apply"? The only issue I have with this method is that it saves the loaded file as a new file in my workspace but I would rather have it my cache and disappear on applying
